Gems/Prefab/PrefabBuilder/PrefabGroup.cpp

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include "PrefabGroup.h"
+#include <AzCore/Serialization/SerializeContext.h>
+#include <AzCore/RTTI/BehaviorContext.h>
+#include <AzCore/JSON/error/error.h>
+#include <AzCore/JSON/error/en.h>
+
+namespace AZ::SceneAPI::SceneData
+{
+     PrefabGroup::PrefabGroup()
+        : m_id(Uuid::CreateNull())
+        , m_name()
+    {
+         m_prefabDom.SetNull();
+    }
+
+    const AZStd::string& PrefabGroup::GetName() const
+    {
+        return m_name;
+    }
+
+    void PrefabGroup::SetName(AZStd::string&& name)
+    {
+        m_name = AZStd::move(name);
+    }
+
+    const Uuid& PrefabGroup::GetId() const
+    {
+        return m_id;
+    }
+
+    void PrefabGroup::SetId(Uuid&& id)
+    {
+        m_id = AZStd::move(id);
+    }
+
+    Containers::RuleContainer& PrefabGroup::GetRuleContainer()
+    {
+        return m_rules;
+    }
+
+    const Containers::RuleContainer& PrefabGroup::GetRuleContainerConst() const
+    {
+        return m_rules;
+    }
+            
+    DataTypes::ISceneNodeSelectionList& PrefabGroup::GetSceneNodeSelectionList()
+    {
+        return m_nodeSelectionList;
+    }
+
+    const DataTypes::ISceneNodeSelectionList& PrefabGroup::GetSceneNodeSelectionList() const
+    {
+        return m_nodeSelectionList;
+    }
+
+    void PrefabGroup::SetPrefabDom(AzToolsFramework::Prefab::PrefabDom&& prefabDom)
+    {
+        m_prefabDom = AZStd::move(prefabDom);
+        m_prefabDomBuffer.clear();
+    }
+
+    void PrefabGroup::SetPrefabDomBuffer(AZStd::string&& prefabDomBuffer)
+    {
+        m_prefabDomBuffer = AZStd::move(prefabDomBuffer);
+        m_prefabDom.SetNull();
+    }
+
+    const AzToolsFramework::Prefab::PrefabDom& PrefabGroup::GetPrefabDom() const
+    {
+        if (!m_prefabDom.HasParseError() && !m_prefabDom.IsNull())
+        {
+            // DOM is ready
+            return m_prefabDom;
+        }
+        else if (m_prefabDomBuffer.empty())
+        {
+            m_prefabDom.SetNull();
+            return m_prefabDom;
+        }
+
+        m_prefabDom.Parse<rapidjson::kParseCommentsFlag>(m_prefabDomBuffer.c_str());
+        if (m_prefabDom.HasParseError())
+        {
+            AZ_Warning(
+                "prefab",
+                false,
+                "JSON error during parsing at offset %zu: %s",
+                m_prefabDom.GetErrorOffset(),
+                rapidjson::GetParseError_En(m_prefabDom.GetParseError()));
+
+            m_prefabDom.SetNull();
+            return m_prefabDom;
+        }
```

Comment:
Aren't there functions from the Prefab system that are better called? There's a bunch of extra processing that needs to happen to load Prefabs correctly from a file which this seems to ignore.

Reply:
True, that will be done in the Prefab Group Behavior to turn a raw JSON DOM into a prefab. This is a data only structure to attempt to be light weight. The JSON might be built up over a few script calls before it is sent on to the builder behavior module.